ledger/src/blockstore_db.rs

```diff
@@ -153,6 +153,12 @@ pub enum BlockstoreError {
     TransactionIndexOverflow,
     #[error("invalid erasure config")]
     InvalidErasureConfig,
+    #[error("last shred index missing")]
+    UnknownLastIndex(Slot),
+    #[error("missing shred")]
+    MissingShred(Slot, u64),
+    #[error("legacy shred")]
+    LegacyShred(Slot, u64),
```

Comment:
I think you need to refer to the parameters in the string otherwise they are not shown when the error is printed/logged.
For example:
```rust
#[error("legacy shred slot {0}, index {1}")]
LegacyShred(Slot, /*index:*/ u64)
```